retrievalmarket/impl/requestvalidation/requestvalidation.go

```diff
@@ -105,6 +106,12 @@ func (rv *ProviderRequestValidator) validatePull(receiver peer.ID, proposal *ret
 		return nil, errors.New("incorrect selector for this proposal")
 	}
 
+	// If the validation is for a restart request, the state is already being
+	// tracked so no further action is required
+	if isRestart {
+		return nil, nil
```

Comment:
because we are not returning a `ErrPause` here, the GS transfer will resume. 

If we restarted before Unsealing, this could fail.
If we restarted after asking for payment but before getting it, we'll wrongly resume here. 
etc etc.

Reply:
So I think that actually returning nil here will not explicitly resume the data transfer:
https://github.com/filecoin-project/go-data-transfer/blob/a175afc88c43158de8b26982e48d9f50b24474d9/impl/receiver.go#L30

Reply:
@dirkmc 

Yes, but the transfer is not explicitly paused here.

This hook is called when Graphsync receives a new Pull request. If we don't explicitly ask Graphsync to pause this new request, it will continue with the transfer. That is why `ValidatePull` currently returns a Pause.

IIUC, returning an `ErrResume` makes sense only when things have already been paused . There's no point in asking GS to resume a req that hasn't been paued.